tests/test_types.py

```diff
@@ -5107,6 +5112,16 @@ class A(BaseModel):
     }
 
 
+def test_transform_schema():
+    ValidateStrAsInt = Annotated[str, TransformSchema(lambda _s: core_schema.int_schema())]
+
+    class Model(BaseModel):
+        x: Optional[ValidateStrAsInt]
+
+    assert Model(x=None).x is None
+    assert Model(x='1').x == 1
```

Comment:
Transformation from a string repr of an int/float/decimal is no longer automatically executed in V2, right? I'm assuming you are overriding this with `TransformSchema(lambda _s: core_schema.int_schema())`?

Also, this test could be rewritten as parameterized if you wanted to go overboard. Perhaps some other edge cases like negative numbers, or floats? (Though would a float get transformed to an int?)